pkg/evaluator/evaluator.go

```diff
@@ -16,12 +16,18 @@ func NewEvaluator(builtins Builtins) *Evaluator {
 }
 
 type Evaluator struct {
+	Stopped  bool
+	Yielder  Yielder
 	print    func(string)
 	builtins map[string]Builtin
 
 	scope *scope // Current top of scope stack
 }
 
+type Yielder interface {
```

Comment:
a comment perhaps?

```
// Yielder abstracts the method of yielding the CPU so that JavaScript/browser
// events get a chance to be processed. Currently (Feb 2023) it seems that you
// can only yield to JS by sleeping for at least 1ms but having that delay is
// not ideal. Other methods of yielding can be explored by implementing a
// different Yielder.
```

Reply:
nice thanks. done.